code/obj/item/cigarette.dm

```diff
@@ -1222,9 +1222,12 @@
 				return
 
 			if (O.reagents.total_volume)
-				O.reagents.trans_to(src, src.reagents.maximum_volume - src.reagents.get_reagent_amount("fuel"))
-				boutput(user, "<span class='notice'>[src] has been refueled.</span>")
-				playsound(src.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+				if ("fuel" in O.reagents.reagent_list)
```

Comment:
```suggestion
				if (O.reagents.has_reagent("fuel"))
```

Reply:
neat, didnt know about that, stole from welding gun.